face/PersonDirectory/python/enroll_multiple_faces_per_person.ipynb

```diff
@@ -0,0 +1,91 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "821406a6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from shared_functions import create_person, add_person_face, identify_faces, create_dynamic_person_group\n",
+    "\n",
+    "# Replace with your Azure Face API subscription key and endpoint\n",
+    "FACE_KEY = os.environ[\"FACE_API_KEY\"]\n",
+    "FACE_ENDPOINT = os.environ[\"FACE_ENDPOINT_URL\"]\n",
+    "folder_path = 'path_to_folder_containing_subfolders_of_images/'\n",
+    "test_image_path = 'path_to_test_image.jpg'\n",
+    "dynamic_person_group_id = \"dynamic_person_group_id\"\n",
+    "\n",
+    "persond_ids = []\n",
+    "# Iterate through all subfolders in the folder_path\n",
+    "for subfolder_name in os.listdir(folder_path):\n",
+    "    subfolder_path = os.path.join(folder_path, subfolder_name)\n",
+    "    if os.path.isdir(subfolder_path):\n",
+    "        person_name = subfolder_name\n",
+    "        # Create a person for each subfolder\n",
+    "        person_id = create_person(FACE_KEY, FACE_ENDPOINT, person_name, \"sample=enroll_multiple_faces_per_person\")\n",
+    "        print(f\"Created person {person_name} with person_id: {person_id}\")\n",
+    "        if person_id:\n",
+    "            persond_ids.append(person_id)\n",
+    "        # Iterate through all images in the subfolder\n",
+    "        for filename in os.listdir(subfolder_path):\n",
+    "            if filename.lower().endswith(('.png', '.jpg', '.jpeg')):\n",
+    "                image_path = os.path.join(subfolder_path, filename)\n",
+    "                # Add face to the created person\n",
+    "                persisted_face_id = add_person_face(FACE_KEY, FACE_ENDPOINT, image_path, person_id, \"sample=enroll_multiple_faces_per_person\")\n",
+    "                print(f\"Added face from {filename} with persisted_face_id: {persisted_face_id} to person_id: {person_id}\")\n",
+    "\n",
+    "print(f\"Person IDs: {persond_ids}\")\n",
+    "# Identify faces in the test image using the person IDs\n",
+    "identify_faces(FACE_KEY, FACE_ENDPOINT, test_image_path, person_ids=persond_ids, injection_header=\"sample=enroll_multiple_faces_per_person\")\n",
+    "\n",
+    "# Create a dynamic person group based on the person IDs\n",
+    "if create_dynamic_person_group(FACE_KEY, FACE_ENDPOINT, dynamic_person_group_id, persond_ids, \"sample=enroll_multiple_faces_per_person\"):\n",
+    "    print(f\"Created dynamic person group with ID: {dynamic_person_group_id}\")\n",
+    "\n",
+    "    # Identify faces in the test image using the dynamic person group\"\"\"\n",
+    "    identify_faces(FACE_KEY, FACE_ENDPOINT, test_image_path, dynamic_person_group_id=dynamic_person_group_id, injection_header=\"sample=enroll_multiple_faces_per_person\")"
```

Comment:
Do we want to clean up the DPG and the person entries at the end of the samples?

Reply:
Good suggestion. Added as the option.